core/utils/promotions/filtered-promotions-manager.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Elementor\Core\Utils\Promotions;
+
+use function DI\string;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Filtered_Promotions_Manager {
+
+	/**
+	 * @param array $promotion_data
+	 * @param string $filter_name
+	 * @param string $url_key
+	 * @return array
+	 */
+	public static function get_filtered_promotion_data( array $promotion_data, string $filter_name, string $url_key, string $url_sub_key = '' ): array {
+		$new_promotion_data = apply_filters( $filter_name, $promotion_data );
+		if ( ! is_array( $new_promotion_data ) ) {
+			return $promotion_data;
+		}
+
+		$new_promotion_data = array_intersect_key( $new_promotion_data, array_flip( array_keys( $promotion_data ) ) );
+
+		$new_promotion_data = self::filter_invalid_url( $new_promotion_data, $url_key, $url_sub_key );
+		return array_replace( $promotion_data, $new_promotion_data );
+	}
+
+	private static function domain_is_on_elementor_dot_com( $url ): bool {
+		$domain = wp_parse_url( $url, PHP_URL_HOST );
+		return isset( $domain ) && str_contains( $domain, 'elementor.com' );
+	}
+
+	private static function filter_invalid_url( $new_promotion_data, string $url_key, string $url_sub_key ) {
```

Comment:
I thought about it
but this increases the level of complexity 
this might also affect how the value is passed to get_filtered_promotion_data and if a change of an API ( even if internal )

IDK

Reply:
@louiswol94 
I don't think this is a blocker, maybe we can fix it in another PR?